### PR TITLE
[CI] Updated the workflows for Rust bindings release

### PR DESCRIPTION
**Pull request by apepkuss**
In this PR, the workflows for Rust bindings release are updated. The original workflows for rustdoc release are removed.

### CODE REVIEW

**Comment by apepkuss** [comment]
@hydai Could you please help review this PR? Thanks a lot!

File: .github/workflows/rust-wasmedge-macro-release.yml [hunk 1]
```diff
@@ -5,62 +5,81 @@ concurrency:
...
-    branches:
-      - "master"
-      - "rust/release-wasmedge-macro"
-    paths:
-      - ".github/workflows/rust-wasmedge-macro-release.yml"
-      - "bindings/rust/wasmedge-macro/**"
-    tags:
-      - "rust-macro/[0-9]+.[0-9]+.[0-9]+*"
-  pull_request:
+  workflow_dispatch:
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-macro-release.yml:8]: Not sure why we give up the release trigger and change to manually trigger?

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/rust-wasmedge-macro-release.yml [hunk 2]
```diff
@@ -5,62 +5,81 @@ concurrency:
...
-        if: github.ref_type == 'branch'
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
[the remark this answers is not in the document]
Comment (apepkuss) [reply .github/workflows/rust-wasmedge-macro-release.yml:44]: Accepted. Will remove it. Thanks!

File: .github/workflows/rust-wasmedge-macro-release.yml (hunk 1 above)
Comment (apepkuss) [reply .github/workflows/rust-wasmedge-macro-release.yml:8]: The goal of the new design of the release workflows is to automate "everything", including adding tags, updating change log, bumping version, and etc.. As a result, triggering the release manually is the only thing we have to do when we wanna publish new crates and API docs. That would extremely reduce the original release job (git tag, git push, etc.) to a mouse click.

Comment (hydai) [reply .github/workflows/rust-wasmedge-macro-release.yml:8]: Got it.

File: .github/workflows/rust-wasmedge-sdk-release.yml [hunk 3]
```diff
@@ -5,75 +5,87 @@ concurrency:
...
 
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sdk-release.yml:44]: Ditto.

File: .github/workflows/rust-wasmedge-sys-release.yml [hunk 4]
```diff
@@ -5,77 +5,87 @@ concurrency:
...
 
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sys-release.yml:44]: Ditto.

File: .github/workflows/rust-wasmedge-types-release.yml [hunk 5]
```diff
@@ -5,62 +5,81 @@ concurrency:
...
-        if: github.ref_type == 'branch'
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-types-release.yml:44]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/rust-wasmedge-sdk-release.yml (hunk 3 above)
Comment (apepkuss) [reply .github/workflows/rust-wasmedge-sdk-release.yml:44]: Accepted. Thanks!

File: .github/workflows/rust-wasmedge-sys-release.yml (hunk 4 above)
Comment (apepkuss) [reply .github/workflows/rust-wasmedge-sys-release.yml:44]: Accepted. Thanks!

File: .github/workflows/rust-wasmedge-types-release.yml (hunk 5 above)
Comment (apepkuss) [reply .github/workflows/rust-wasmedge-types-release.yml:44]: Accepted. Thanks!

**Review by hydai — APPROVED** [review]

**Comment by apepkuss** [comment]
@hydai The issues have been fixed. Thanks!

**Comment by apepkuss** [comment]
@hydai Thanks for the review!

**Pull request merged**